### PR TITLE
SAK-44171 Lessons tool does not display Resource folder displayname

### PR DESCRIPTION
**Pull request by austin48**

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java [hunk 1]
```diff
@@ -3096,6 +3097,19 @@ public int compare(SimpleStudentPage o1, SimpleStudentPage o2) {
 						String folderName = folderPath[folderPath.length-1];
 						if (dataDirectory.endsWith("//")){
 							folderName = simplePageBean.getCurrentSite().getTitle();
+						} else {
+							try {
+								ContentHostingService contentHostingService = (ContentHostingService) ComponentManager.get("org.sakaiproject.content.api.ContentHostingService");
+								ContentCollection collection = contentHostingService.getCollection(dataDirectory.replace("//", "/"));
+								if (collection != null) {
+									String displayName = collection.getProperties().getProperty(collection.getProperties().getNamePropDisplayName());
+									if (displayName != null && displayName != "") {
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:3106]: Why not just use ResourceProperties.PROP_DISPLAY_NAME?
Also, you could use StringUtils.isBlank  on the displayName test.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java [hunk 2]
```diff
@@ -3096,6 +3097,19 @@ public int compare(SimpleStudentPage o1, SimpleStudentPage o2) {
...
+										folderName = displayName;
+									}
+								}
+							} catch (Exception e) {
+								log.error("Error getting content resource" + e);
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:3111]: You should probably use log.error("error message", e) here.

**Review by adrianfish — CHANGES_REQUESTED** [review]
A couple of small things.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java (hunk 1 above)
Comment (austin48) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java]: fixed

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java (hunk 2 above)
Comment (austin48) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java]: fixed.  thanks for the suggestions

**Comment by ern** [comment]
Austin I've refactored your PR can you click the option "Allow edits from maintainers" so that i can push these changes to your branch?

**Comment by austin48** [comment]
> Austin I've refactored your PR can you click the option "Allow edits from maintainers" so that i can push these changes to your branch?

It looks like that option is already checked

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java [hunk 3]
```diff
@@ -4289,18 +4286,9 @@ private void createToolBar(UIContainer tofill, SimplePage currentPage) {
 
 		    // add website.
 		    // Are we running a kernel with KNL-273?
-		    Class contentHostingInterface = ContentHostingService.class;
-		    try {
-			Method expandMethod = contentHostingInterface.getMethod("expandZippedResource", new Class[] { String.class });
-			
 			UIOutput.make(tofill, "addwebsite-li");
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:4289]: Can you outdent these two lines?

**Review by adrianfish — APPROVED** [review]
GTG

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java (hunk 3 above)
Comment (ern) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:4289]: yeah the indentation is mixed here, lines are using 2 tabs and 4 spaces so decided not to update it to keep history

Comment (austin48) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:4289]: Thanks for refactoring @ern!

I'd like to backport your refactor to our 12.7 instance.  Is there anything I'd need to adjust from your backport?

**Comment by austin48** [comment]
Thanks for refactoring @ern!

I'd like to backport your refactor to our 12.7 instance. Is there anything I'd need to adjust from your backport?

**Comment by ottenhoff** [comment]
> `@Setter ContentHostingService contentHostingService;`
> 
> Should this be a private variable?

In Java, the default Access Modifier for all the members of class is Package Private. That means, if no type is defined for any variable at the time of declaration, it is package private. That variable can only be accessed by all classes or members inside that package.

**Comment by austin48** [comment]
ok, thanks

**Pull request merged**